src/uri/authority.rs

```diff
@@ -53,6 +53,7 @@ impl Authority {
         })
     }
 
+    // Note: this may return an *empty* Authority. You might want `parse_non_empty`.
     pub(super) fn parse(s: &[u8]) -> Result<usize, InvalidUri> {
```

Comment:
This looks to be crate private. Is there a reason the caller could pass in an empty authority?

Reply:
Yes, there is, in `parse_full` for `Uri`. An empty authority is handled differently there (error is `MissingAuthority` instead of `Empty`).